Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -153,8 +153,12 @@ namespace PhysX
     void EditorHeightfieldColliderComponent::BuildGameEntity(AZ::Entity* gameEntity)
     {
         auto* heightfieldColliderComponent = gameEntity->CreateComponent<HeightfieldColliderComponent>();
+
+        // Create an empty shapeConfig, it will get filled out at runtime as everything initializes.
+        AZStd::shared_ptr<Physics::HeightfieldShapeConfiguration> shapeConfig{ new Physics::HeightfieldShapeConfiguration() };
+
         heightfieldColliderComponent->SetShapeConfiguration(
-            { AZStd::make_shared<Physics::ColliderConfiguration>(m_colliderConfig), m_shapeConfig });
+            { AZStd::make_shared<Physics::ColliderConfiguration>(m_colliderConfig), shapeConfig });
```

Comment:
question: If `m_shapeConfig` is not used and it will calculated at runtime, does this editor component still need to have `m_shapeConfig` member and reflected in serialize context?

Reply:
Good question. I'm not quite sure, since I was a bit surprised it wasn't getting used. I'll leave that question for a separate investigation and PR. I only made this change because there was some bad data leaking into unit tests due to the way they mock up so many systems, including a subverted means of creating game entities. I verified the change _works_ in both unit tests and actual Editor usage, but I didn't investigate further to see exactly what was going on. The short version of what I saw is that in the Editor, the m_shapeConfig was cleared out at some point before creating the game entity, but in unit tests, it wasn't, so bad native heightfield pointers were getting attached to the game entity in the unit tests. This prevented the pointer leak and caused consistent behavior to happen between game exports and unit tests.

Reply:
Thanks @mbalfour-amzn . Yeah, let's investigate further and see if it can be removed. Tagging @SergeyAMZN in case he has more information.

> I'll leave that question for a separate investigation and PR.
:+1:

> I verified the change works in both unit tests and actual Editor usage
That's good. Before submitting this version, just for extra precaution, I would also recommend running the launcher with a level containing these components and see if they are handled correctly.

Reply:
Verified that the heightfield physics is behaving correctly in the launcher.

Reply:
Thanks for checking.

Reply:
Yes, there's nothing serialized out of the shape configuration. 
We get the actual info about heights & materials from the terrain physics component even at game time. 

I'm working on having on option of baking the heightfield into an asset, so there will be more use of the serialized shape configuration